src/org/labkey/test/tests/SampleTypeTest.java

```diff
@@ -713,32 +698,47 @@ public void testUpdateAndDeleteWithCommentsAndFlags()
 
         updateSampleType(sampleData.get(testDataIndex));
 
-        errorMsg = "Sample Type data is not as expected after a updating both a Description and a Flag/Comment.";
-        if(!checkExpectedAgainstDB(sampleData, errorMsg))
-        {
-            errorLog.append("Failure with 'updating both a description and flag/comment' test.\n");
-            errorLog.append(errorMsg);
-            errorLog.append("\n");
-        }
+        resultsFromDB = getSampleDataFromDB("/" + PROJECT_NAME,SAMPLE_TYPE_NAME, dbFieldsToCheck);
+        checker().verifyTrue("Sample Type data is not as expected after a updating both a Description and a Flag/Comment.",
+                areDataListEqual(resultsFromDB, sampleData));
 
         log("Now delete both the Description and Flag/Comment from the sample.");
         sampleData.get(testDataIndex).replace("Flag", "");
         sampleData.get(testDataIndex).replace("Description", "");
 
         updateSampleType(sampleData.get(testDataIndex));
 
-        errorMsg = "Sample Type data is not as expected after deleting the Description and Flag/Comment.";
-        if(!checkExpectedAgainstDB(sampleData, errorMsg))
+        resultsFromDB = getSampleDataFromDB("/" + PROJECT_NAME,SAMPLE_TYPE_NAME, dbFieldsToCheck);
```

Comment:
Would `getCurrentContainerPath()` return this catenated projectName and "/"?

Reply:
Yes it does.
Removed all the `"/" + PROJECT_NAME` and replaced with `getCurrentContainerPath()`.